### PR TITLE
Aadding a test for cancelable context

### PR DESCRIPTION
**Pull request by vwiart**
_No description provided._

### CODE REVIEW

File: client/client_test.go [hunk 1]
```diff
@@ -196,3 +196,26 @@ func TestRetries(t *testing.T) {
...
+
+	ch := make(chan error)
+	go func() {
+		cli, _ := client.New()
+		_, err := cli.Put(ctx, server.URL)
+		ch <- err
+	}()
+
+	cancel()
+	err := <-ch
```
Comment (voroninman) [inline client/client_test.go:216]: This will hang if there is a bug. Don't you think we need to wrap it in a select with some timeout?

**Review by voroninman — CHANGES_REQUESTED** [review]
I'm ready to approve but one thing. Glad to see tests! 👍😍

File: client/client_test.go (hunk 1 above)
Comment (vwiart) [reply client/client_test.go:216]: [tests automatically timeout after a while](https://golang.org/cmd/go/#hdr-Testing_flags)

```
-timeout d
    If a test binary runs longer than duration d, panic.
    If d is 0, the timeout is disabled.
    The default is 10 minutes (10m).
```

Comment (voroninman) [reply client/client_test.go:216]: Are you saying that this is fine to wait 10m for a test to fail? 😀

Comment (vwiart) [reply client/client_test.go:216]: I think an easy fix would be to Ctrl-C the test :)

Keep in mind this is a library, there are no Rakefile to build nor to test. Test are only done in the IDE if the developer implementing a feature think about it.

**Review by nlundbo — APPROVED** [review]

**Pull request merged**